code/modules/mob/living/silicon/pai/software_modules.dm

```diff
@@ -547,3 +547,58 @@
 			P.sradio.code = min(100, P.sradio.code)
 			P.sradio.code = max(1, P.sradio.code)
 			return 1
+
+/datum/pai_software/host_scan
+	name = "Host Bioscan"
+	ram_cost = 5
+	id = "bioscan"
+	toggle = 0
+
+
+	on_ui_interact(mob/living/silicon/pai/user, datum/nanoui/ui=null, force_open=1)
+
+		var/data[0]
+		var/mob/living/held = user.loc
+		var/count = 0
+
+			// Find the carrier
+		while(!istype(held, /mob/living))
+			if(!held || !held.loc || count > 6)
+				//For a runtime where M ends up in nullspace (similar to bluespace but less colourful)
+				src << "You are not being carried by anyone!"
+				return 0
+			held = held.loc
+			count++
+		if(istype(held, /mob/living))
+			data["holder"] = held
+			data["health"] = "[held.stat > 1 ? "dead" : "[held.health]% healthy"]"
+			data["oxy"] = "[held.getOxyLoss() > 50 ? "<font color=#FF5555>" : "<font color=#55FF55>"][held.getOxyLoss()]</font>"
+			data["tox"] = "[held.getToxLoss() > 50 ? "<font color=#FF5555>" : "<font color=#55FF55>"][held.getToxLoss()]</font>"
+			data["burn"] = "[held.getFireLoss() > 50 ? "<font color=#FF5555>" : "<font color=#55FF55>"][held.getFireLoss()]</font>"
+			data["temp"] = "[held.bodytemperature-T0C]&deg;C ([held.bodytemperature*1.8-459.67]&deg;F)"
+			//data["infected"] = null
+			if (held.viruses)
+				data["infected"] = 1
+				var/virus[0]
+				for((var/datum/disease2/disease/D in held.viruses))
```

Comment:
I'm pretty sure there should be a check somewhere in here to compare to the virology records before showing all of it's data.